pkg/gtid/gtid.go

```diff
@@ -47,16 +48,35 @@ type Set interface {
 // ParserGTID parses GTID from string
 func ParserGTID(flavor, gtidStr string) (Set, error) {
 	var (
-		m   Set
-		err error
+		m    Set
+		err  error
+		gtid mysql.GTIDSet
 	)
 
-	gtid, err := mysql.ParseGTIDSet(flavor, gtidStr)
+	if len(flavor) == 0 && len(gtidStr) == 0 {
+		return nil, errors.Errorf("empty flavor with empty gtid is invalid")
+	}
+
+	fla := flavor
```

Comment:
```
if len(flavor) == 0 {
    flavor = mysql.MySQLFlavor
}
gtid, err = mysql.ParseGTIDSet(flavor, gtidStr)
if err != nil {
    flavor = mysql. MariaDBFlavor
    gtid, err = mysql.ParseGTIDSet(flavor, gtidStr)
    if err != nil {
        return nil, terror.ErrNotSupportedFlavor.Generate(flavor)
    }
}

Reply:
If user set `flavor: MySQL` but use MariaDB as upstream,  we may need to report an error?

Reply:
> If user set flavor: MySQL but use MariaDB as upstream, we may need to report an error?

I agress that we should report an error, so we should differ empty flavor and user specified `mysql` flavor when  `mysql.ParseGTIDSet("mysql", gtidStr)` failed. PTAL @3pointer